java/src/hu/holyoil/resource/Uranium.java

```diff
@@ -36,7 +51,12 @@ public Boolean IsSameType(AbstractBaseResource abstractBaseResource) {
     @Override
     public void ReactToSunNearby(Asteroid asteroid) {
         Logger.Log(this,"Reacting to Sun nearby");
-        asteroid.Explode();
+        
+        SetHealth(health-1);
+
+        if(health <= 0)
```

Comment:
This is currently inconsistent with seq diagram.

Reply:
tru dat